### PR TITLE
Further `unstable` Cleanup

### PR DESCRIPTION
**Pull request by stamourv**
Tackled a few more subcollects of `unstable`.

* Merged `unstable/future` with `racket/future`.
* Merged `unstable/open-place` with `racket/place`.
* Merged part of `unstable/logging` to a new `racket/logging`, and improve interface a bit.
* Decided against merging `unstable/match`. `match?` would conflict with a different function of the same name from Redex, and `object` would conflict with exports from `html` and `scribble/html`. The plan is to move `unstable/match` out of the TR repo and into `unstable-list-lib`.
* Decided against merging `unstable/string`.

### CODE REVIEW

**Comment by rfindler** [comment]
The futures part looks great to me.

File: pkgs/racket-doc/scribblings/reference/places.scrbl [hunk 1]
```diff
@@ -262,6 +262,14 @@ The @racket[place] binding is protected in the same way as
 The @racket[place*] binding is protected in the same way as
  @racket[dynamic-place].}
 
+@defform[(open-place id body ...+)]{
+  Like @racket[place], but @racket[body ...] may have free lexical
+  variables, which are automatically sent to the newly-created place.
+  Note that these variables must have values accepted by
+  @racket[place-message-allowed?], otherwise an @racket[exn:fail:contract]
```
Comment (samth) [inline pkgs/racket-doc/scribblings/reference/places.scrbl]: There's an abstraction for this last clause somewhere in the reference.

**Comment by stamourv** [comment]
@samth: Thanks for the pointer! Fixed locally. Any other feedback?

**Comment by samth** [comment]
Looks good.

**Comment by stamourv** [comment]
Merged.

**Pull request closed without merging**